muxtools/subtitle/font.py

```diff
@@ -48,6 +55,200 @@ def _parse_unicode_chars(char_list: list[str]) -> list[str]:
+        total_old_size += old_size
+
+        font_name = _get_fontname(font_face)
+
+        ttLib_font = ttLib.TTFont(font_face.font_file.filename)
+
+        name_table = ttLib_font["name"]
+
+        for record in name_table.names:
+            if record.nameID == 1:  # Font Family name
+                old_name = record.toUnicode().strip()
+                if old_name in data["names"]:
+                    record.string = data["names"][old_name]
+            
+            elif record.nameID == 4:  # Full name
+                old_name = record.toUnicode().strip()
+                if old_name in data["names"]:
+                    record.string = data["names"][old_name]
+            
+            elif record.nameID == 6:  # PostScript name
+                old_name = record.toUnicode().strip()
+                if old_name in data["names"]:
+                    record.string = data["names"][old_name]
+        
+        characters = data["usage"].copy()
+        characters.update(subset_additional_glyphs_parsed)
+
+        if not aggressive:
+            # We also want to add a general subset of common characters to avoid issues when reusing fonts, if we aren't doing aggressive subsetting
+            characters.update(COMMON_UNICODE_CHARS)
+
+        if not characters:
+            if ignore_fonts_with_no_usage:
+                warn(f"No characters used in font '{font_name}'. Skipping subsetting.", subset_fonts)
+                continue
+            else:
+                warn(f"No characters used in font '{font_name}'. Defaulting to common subset.", subset_fonts)
+                characters.update(COMMON_UNICODE_CHARS)
+        
+        subsetter = Subsetter()
+        subsetter.populate(text="".join(characters))
+        subsetter.subset(ttLib_font)
+
+        new_font_path = font_face.font_file.filename.with_name(f"{font_face.font_file.filename.stem}_subset{font_face.font_file.filename.suffix}")
+        ttLib_font.save(new_font_path)
+        ttLib_font.close()
+
+        new_size = os.path.getsize(new_font_path)
+        total_new_size += new_size
+
+        os.remove(font_face.font_file.filename)
+        
+        for old_name, new_name in data["names"].items():
+            font_replacements[old_name] = new_name
+
+        info(f"Subsetted font '{font_name}' ({len(characters)} glyphs, {sizeof_fmt(old_size)} -> {sizeof_fmt(new_size)})", collect_fonts)
+    
+
+    if font_replacements:
+        for sub in subs:
+            doc = sub._read_doc()
+
+            for event in doc.events:
+                line_data = parse_line(event.text)
+                for data in line_data:
+                    if isinstance(data, AssValidTagFontName):
+                        safe_font_name = re.escape(data.name)
+
+                        for old_font_name, new_font_name in font_replacements.items():
+                            if data.name == old_font_name:
+                                event.text = re.sub(
```

Comment:
you could maybe instead directly modify the `AssValidTagFontName`, and rebuild the line from a modified `line_data`.

this *does* hinge on the assumption that `parse_line` always parses things correctly, though (notably, it will perform some “canonicalization”, e.g. `\pos( 1 ,   4)` would become `\pos(1,4)`, but as long as `ass_tag_analyzer` is correct, the output will be identical)